### ISSUE TITLE
Bug: jax.lax.switch

### ISSUE THREAD

**Issue by Nik-SteinFraunh** [eclipse-qrisp/Qrisp#192]
The following code does not compile in jasp mode, but it does compile in "normal" @jax.jit mode. it seems that the lax.switch is the problem. Link to error message screenshot at the bottom 

```import networkx as nx 
from qrisp import control, rx, invert, QuantumBool, x, mcx, QuantumVariable, qswitch
from qrisp.jasp import jaspify, jrange, quantum_kernel
import numpy as np
import jax.numpy as jnp 
import jax
""" 
n_nodes = 6
G = nx.erdos_renyi_graph(n_nodes, 0.7, seed = 133)

neighbors_dict = {node: list(G.adj[node]) for node in G.nodes()} """

neighbors_dict={1:[2,3],3:[2,1],0:[1,3],2:[0,3]}

@jaspify#@jax.jit
def main_hh():
    # Compute max length for padding
    max_len = max(len(val) if isinstance(val, list) else 1 for val in neighbors_dict.values())
    
    padded_data = []
    lengths = []
    for i in range(len(neighbors_dict)):
        val = neighbors_dict[i]
        arr = jnp.array(val) if isinstance(val, list) else jnp.array([val])
        ll= arr.shape[0]
        for ii in range(max_len - arr.shape[0] ):
            arr =jnp.append(arr, jnp.array([99])) 

        print(arr)
        padded_data.append(lambda arr=arr: arr)
        lengths.append(lambda l=ll: jnp.array(l))
    def get_value_and_length(i):
        data = jax.lax.switch(i, padded_data)
        length = jax.lax.switch(i, lengths)
        return data, length
    n_nodes = 6

    neighb_array, neigb_len = get_value_and_length(2)
    #ctrl_str = ""
    #jax.debug.print("{neighb_array} neighb_array", neighb_array=neighb_array)
    ctrls = jnp.array([neighb_array[0]])

    return 1# qv


test = main_hh()`

![Image](https://github.com/user-attachments/assets/c750ad1e-c842-4eef-90dc-d33b0acb04f7)

**Comment by positr0nium** [comment]
Fixed in 387c8a8cfd9875e942967814771357360cb11154.
